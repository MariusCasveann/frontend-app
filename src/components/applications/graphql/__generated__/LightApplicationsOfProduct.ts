/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: LightApplicationsOfProduct
// ====================================================

export interface LightApplicationsOfProduct_lightApplicationsOfProduct {
    __typename: 'LightApplicationDTO';
    id: any | null;
    name: string | null;
    service: boolean;
    staticPictureId: any | null;
}

export interface LightApplicationsOfProduct {
    lightApplicationsOfProduct: (LightApplicationsOfProduct_lightApplicationsOfProduct | null)[] | null;
}

export interface LightApplicationsOfProductVariables {
    productId?: any | null;
}
