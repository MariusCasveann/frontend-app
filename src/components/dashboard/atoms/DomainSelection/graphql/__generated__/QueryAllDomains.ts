/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: QueryAllDomains
// ====================================================

export interface QueryAllDomains_allDomains {
    __typename: 'Domain';
    id: any | null;
    name: string | null;
}

export interface QueryAllDomains {
    allDomains: (QueryAllDomains_allDomains | null)[] | null;
}
