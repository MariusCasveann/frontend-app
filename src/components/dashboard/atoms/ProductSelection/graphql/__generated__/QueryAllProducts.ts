/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: QueryAllProducts
// ====================================================

export interface QueryAllProducts_allProducts {
    __typename: 'LightProductDTO';
    id: any | null;
    logo: string | null;
    name: string | null;
}

export interface QueryAllProducts {
    allProducts: (QueryAllProducts_allProducts | null)[] | null;
}
