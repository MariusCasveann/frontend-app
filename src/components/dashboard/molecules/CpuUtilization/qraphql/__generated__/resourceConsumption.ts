/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {
    DeploymentStageEnum,
    ResourceTypeEnum,
    ResourceAggregationTypeEnum,
    QueryEntity
} from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: ResourceConsumption
// ====================================================

export interface ResourceConsumption_resourceConsumption {
    __typename: 'ResourceConsumptionDTO';
    projectId: string | null;
    resourceType: string | null;
    usage: string | null;
    usageDate: any | null;
}

export interface ResourceConsumption {
    resourceConsumption: (ResourceConsumption_resourceConsumption | null)[] | null;
}

export interface ResourceConsumptionVariables {
    entityId?: any | null;
    deploymentStage?: DeploymentStageEnum | null;
    resourceType?: ResourceTypeEnum | null;
    aggregationType?: ResourceAggregationTypeEnum | null;
    entityType?: QueryEntity | null;
}
