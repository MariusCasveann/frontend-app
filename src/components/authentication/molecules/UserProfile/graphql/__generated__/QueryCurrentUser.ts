/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { PermissionTypeEnum, SecuredEntityTypeEnum } from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: QueryCurrentUser
// ====================================================

export interface QueryCurrentUser_currentUser_roles_concretePermissions {
    __typename: 'EntityPermission';
    permissionType: PermissionTypeEnum | null;
    id: any | null;
    entityType: SecuredEntityTypeEnum | null;
    entityId: any | null;
}

export interface QueryCurrentUser_currentUser_roles {
    __typename: 'Role';
    id: any | null;
    name: string | null;
    readOnly: boolean | null;
    concretePermissions: (QueryCurrentUser_currentUser_roles_concretePermissions | null)[] | null;
}

export interface QueryCurrentUser_currentUser {
    __typename: 'UserDTO';
    activated: boolean;
    createdBy: string | null;
    createdDate: any | null;
    email: string | null;
    firstName: string | null;
    id: any | null;
    imageContent: string | null;
    langKey: string | null;
    lastModifiedBy: string | null;
    lastModifiedDate: any | null;
    lastName: string | null;
    login: string | null;
    roles: (QueryCurrentUser_currentUser_roles | null)[] | null;
    staticPictureId: any | null;
    subscribedToNewsletter: boolean | null;
}

export interface QueryCurrentUser {
    currentUser: QueryCurrentUser_currentUser | null;
}
