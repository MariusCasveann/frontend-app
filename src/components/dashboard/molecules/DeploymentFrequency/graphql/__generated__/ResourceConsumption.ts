/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { ResourceTypeEnum, QueryEntity } from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: ResourceConsumption
// ====================================================

export interface ResourceConsumption_resourceConsumption {
    __typename: 'ResourceConsumptionDTO';
    projectId: string | null;
    usage: string | null;
}

export interface ResourceConsumption {
    resourceConsumption: (ResourceConsumption_resourceConsumption | null)[] | null;
}

export interface ResourceConsumptionVariables {
    entityId?: any | null;
    resourceType?: ResourceTypeEnum | null;
    entityType?: QueryEntity | null;
    toDate?: any | null;
    fromDate?: any | null;
}
