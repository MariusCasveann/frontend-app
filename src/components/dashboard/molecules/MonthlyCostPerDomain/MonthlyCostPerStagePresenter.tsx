import React from 'react';
import NumberDisplay from '../../../common/molecules/NumberDisplay/NumberDisplay';
import { DeploymentStageEnum } from '../../../../model/__generated__/globalTypes';
import LoadingOrErrorPresenter from '../../../common/organisms/LoadingOrError/LoadingOrErrorPresenter';
import './MonthlyCostPerDomain.css';
import { MESSAGE_WHEN_NO_DATA } from '../../../../utils/constants';

interface MonthlyCostPerStageProps {
    currency: string;
    error: string;
    loading: boolean;
    stage: DeploymentStageEnum;
    value: number;
}

export default ({ currency, stage, value, error, loading }: MonthlyCostPerStageProps) => {
    if (loading || error) {
        return <LoadingOrErrorPresenter error={error} loading={loading} />;
    }
    if (value > 0) {
        return <NumberDisplay currency={currency} value={value} label={stage} />;
    }
    return <p>{MESSAGE_WHEN_NO_DATA}</p>;
};
