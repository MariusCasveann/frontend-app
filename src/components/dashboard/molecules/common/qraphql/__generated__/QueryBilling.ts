/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { QueryEntity } from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: QueryBilling
// ====================================================

export interface QueryBilling_billingQuery {
    __typename: 'GoogleBillingResultDTO';
    resourceType: string | null;
    cost: any | null;
    date: any | null;
    currency: string | null;
}

export interface QueryBilling {
    billingQuery: (QueryBilling_billingQuery | null)[] | null;
}

export interface QueryBillingVariables {
    entity?: QueryEntity | null;
    entityId?: number | null;
    stage?: string | null;
    financialYear?: number | null;
    toDate?: any | null;
    fromDate?: any | null;
}
