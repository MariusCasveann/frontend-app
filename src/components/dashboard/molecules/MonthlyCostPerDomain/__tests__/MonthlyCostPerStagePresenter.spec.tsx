import * as React from 'react';
import renderer from 'react-test-renderer';
import { DeploymentStageEnum } from '../../../../../model/__generated__/globalTypes';
jest.mock('../../../../common/organisms/LoadingOrError/LoadingOrErrorPresenter', () => 'LoadingOrErrorPresenter');
jest.mock('../../../../common/molecules/NumberDisplay/NumberDisplay', () => 'NumberDisplay');
import MonthlyCostPerStagePresenter from '../MonthlyCostPerStagePresenter';

const mockCurrency = 'EUR';

describe('MonthlyCostPerStagePresenter', () => {
    describe('when there is value', () => {
        it('renders a NumberDisplay', () => {
            const wrapper = renderer
                .create(
                    <MonthlyCostPerStagePresenter
                        currency={mockCurrency}
                        error=""
                        loading={false}
                        value={78}
                        stage={DeploymentStageEnum.DEV}
                    />
                )
                .toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when there is zero value', () => {
        it('renders any data', () => {
            const wrapper = renderer
                .create(
                    <MonthlyCostPerStagePresenter
                        currency={mockCurrency}
                        error=""
                        loading={false}
                        value={0}
                        stage={DeploymentStageEnum.DEV}
                    />
                )
                .toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
