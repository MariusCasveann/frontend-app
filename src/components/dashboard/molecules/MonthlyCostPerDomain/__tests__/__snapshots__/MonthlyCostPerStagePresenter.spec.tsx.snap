// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyCostPerStagePresenter when there is value renders a NumberDisplay 1`] = `
<NumberDisplay
  currency="EUR"
  label="DEV"
  value={78}
/>
`;

exports[`MonthlyCostPerStagePresenter when there is zero value renders any data 1`] = `
<p>
  No data available, please select another time and/or domain/product.
</p>
`;
